### ISSUE TITLE
pyo3_runtime.PanicException on latest 0.18.9

### ISSUE THREAD

**Issue by qqletter** [pola-rs/polars#10196]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
        df_trend = (
            df_trend.lazy()
            .with_columns([
                pl.col("trade_date").cast(pl.Date),
                pl.when(pl.col("change").over('combo_code') > 0).then(1).otherwise(0).alias("change"),
            ])
            .select([
                pl.all(),
                pl.col("change").rolling_sum(window_size=5, closed='right').over("combo_code").fill_null(0).cast(pl.Int16).alias("up_in_5days"),
                pl.col("change").rolling_apply(lambda x: self._rising_count(x), window_size=11, min_periods=1).over("combo_code").fill_null(0).cast(pl.Int16).alias("up_days"),
                (pl.col("close")-pl.col("close").shift(2)).arctan().degrees().over('combo_code').fill_null(0).cast(pl.Float32).round(4).alias("slope_3"),
                (pl.col("close")-pl.col("close").shift(4)).arctan().degrees().over('combo_code').fill_null(0).cast(pl.Float32).round(4).alias("slope_5"),
                (pl.col("close")-pl.col("close").shift(6)).arctan().degrees().over('combo_code').fill_null(0).cast(pl.Float32).round(4).alias("slope_7"),
                (pl.col("close")-pl.col("close").shift(9)).arctan().degrees().over('combo_code').fill_null(0).cast(pl.Float32).round(4).alias("slope_10"),
                pl.col("ma5_10").arctan().degrees().over('combo_code').fill_null(0).cast(pl.Float32).round(4).alias("slope_ma5"),
            ])
            .filter(pl.col("trade_date") >= max_date)
            .collect()
        )

```

### Issue description

 pyo3_runtime.PanicException: called `Result::unwrap()` on an `Err` value: PoisonError { .. }
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/execution/plan/execute_plan.py", line 273, in dagster_event_sequence_for_step
    for step_event in check.generator(step_events):
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/execution/plan/execute_step.py", line 369, in core_dagster_event_sequence_for_step
    for user_event in check.generator(
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/execution/plan/execute_step.py", line 90, in _step_output_error_checked_user_event_sequence
    for user_event in user_event_sequence:
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/execution/plan/compute.py", line 192, in execute_core_compute
    for step_output in _yield_compute_results(step_context, inputs, compute_fn):
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/execution/plan/compute.py", line 161, in _yield_compute_results
    for event in iterate_with_context(
  File "/usr/local/lib/python3.11/site-packages/dagster/_utils/__init__.py", line 445, in iterate_with_context
    next_output = next(iterator)
                  ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/execution/plan/compute_generator.py", line 124, in _coerce_op_compute_fn_to_iterator
    result = invoke_compute_fn(
             ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/execution/plan/compute_generator.py", line 118, in invoke_compute_fn
    return fn(context, **args_to_pass) if context_arg_provided else fn(**args_to_pass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/dagster/quantrade_data/ops/transform/stock.py", line 70, in stock_trend_daily
    stk_handler.stock_trend_daily()
  File "/dagster/quantrade_data/handle/stock.py", line 837, in stock_trend_daily
    .collect()
     ^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/polars/utils/decorators.py", line 37, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/polars/lazyframe/frame.py", line 1552, in collect
  return wrap_df(ldf.collect())

### Expected behavior

These code will get an exception as above, when use the latest version 0.18.9.  
If I fallback to verion 0.18.8, the code works well.

### Installed versions

<details>

```
version 0.18.9
```

</details>

**Comment by ritchie46** [comment]
We cannot reproduce without data, but I think we already fixed this one. Patch coming up in an hour, maybe 2.

**Comment by qqletter** [comment]
> We cannot reproduce without data, but I think we already fixed this one. Patch coming up in an hour, maybe 2.

**Comment by qqletter** [comment]
I closed this issue by mistake, so reopen it.

**Comment by alexander-beedie** [comment]
@qqletter: likely fixed in the just-released `0.18.10` - could you try it out and confirm?
